edx_event_bus_kafka/internal/consumer.py

```diff
@@ -169,6 +169,19 @@ def reset_offsets(consumer, partitions):
             logger.error("Kafka consumers not enabled, exiting.")
             return
 
+        # .. setting_name: EVENT_BUS_KAFKA_CONSUMER_CONSECUTIVE_ERRORS_LIMIT
+        # .. setting_default: None
+        # .. setting_description: If the consumer encounters this many consecutive errors, exit with an
```

Comment:
First, I was wondering if we should document the lost connection issue, since this is what we were proposing as the solution. Next, I was wondering what database config solutions there might be, and whether that was looked into, for those that don't want to auto-restart.

Here are the latest Django docs for db: https://docs.djangoproject.com/en/4.1/ref/databases/#persistent-connections. An interesting quote:
> The default value is 0, preserving the historical behavior of closing the database connection at the end of each request. To enable persistent connections, set [CONN_MAX_AGE](https://docs.djangoproject.com/en/4.1/ref/settings/#std-setting-CONN_MAX_AGE) to a positive integer of seconds. For unlimited persistent connections, set it to None.

This got me wondering if we should just force close the connection, which got me to this stackoverlow answer. Not sure if `connect()` is correct, but it could be. See https://stackoverflow.com/a/50694172/13500961

Not sure if we want both these changes (fix db connection and exit after error limit).

Reply:
DB connection config would be outside of the scope of event-bus-kafka, but we may want to mention it in our Event Bus docs. And certainly we should consider trying that `connect()` change in the Discovery workers.

Reply:
Since we're implementing with Django, and implementing an infinite loop (which is different than a request based model), it feels like we'd want to provide reasonable behavior by default if we could. That said, I'm not sure what "reasonable behavior" should be. We have `CONN_MAX_AGE` set to 60, so the above comment shouldn't apply. But, I'm not sure what should happen, because our connection stayed broken beyond 1 minute. Does that mean it requires a `connect()` call to fix the connection, and that isn't happening without a new request?

When one writes a new Django view, this is not something they need to think about, whether the view uses or doesn't use the database. How does that work, and could we replicate that behavior by default (mimicking what Django does)? I don't like the idea that every new signal handler has to worry about this.

Reply:
This being said, I'm now not sure how to best document or fix from a db connection perspective, so maybe we just land this and ticket the rest as an open issue that can sit in future, given that we have this backstop.

Reply:
Hmm... yeah, a close reading of the docs there indicates that the max age is only checked at request start, and usability is only checked at the end of the request, and (I think) a new connection is always only provided at the start of a request, in any case.

I think what we want to do is try various approaches in the discovery-worker management command code, and then determine if the fix can be lifted out—and to where. First I'd just like to get this generic backstop out there, though.